streams/src/main/java/org/apache/kafka/streams/kstream/internals/WindowedStreamPartitioner.java

```diff
@@ -32,6 +32,11 @@
         this.serializer = serializer;
     }
 
+    WindowedStreamPartitioner(final WindowedSerializer<K> serializer) {
+        this.topic = null;
```

Comment:
Do we need an overload for `partition` below with an additional `topic` parameter to make sure that the key serializer get's called with a proper topic name instead of `null` ?

Reply:
I've thought about it before, but this is a bit tricky. Let me try to have some "ugly" solution and let's see if it is okay.

Reply:
I see -- should we do a KIP for this to fix it? We can just create a JIRA for now -- maybe somebody wants to pick it up.